### PR TITLE
GEOMESA-967 Users should be able to build GeoMesa with Accumulo 1.5 support

### PR DESCRIPTION
**Pull request by jdk2pq**
Signed-off-by: Jake Kenneally <jdk2pq@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala [hunk 1]
```diff
@@ -80,9 +90,8 @@ object AccumuloVersion extends Enumeration {
     if (dot > 0) {
       require(accumuloVersion != V15, s"Table namespaces are not supported in Accumulo 1.5 - for table '$table'")
       val ns = table.substring(0, dot)
-      val nsOps = classOf[Connector].getMethod("namespaceOperations").invoke(connector)
-      val nsOpsClass = Class.forName("org.apache.accumulo.core.client.admin.NamespaceOperations")
-      if (!nsOpsClass.getMethod("exists", classOf[String]).invoke(nsOps, ns).asInstanceOf[Boolean]) {
+      val (nsOps, nsOpsClass) = getNsOps(connector)
+      if (!nameSpaceExists(nsOps, nsOpsClass, ns)) {
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala]: could you just do (nsOps, nsOps.getClass, ns) and avoid passing the class back?

File: geomesa-process/pom.xml [hunk 2]
```diff
@@ -22,12 +22,14 @@
         <dependency>
             <groupId>org.apache.accumulo</groupId>
             <artifactId>accumulo-core</artifactId>
-            <scope>provided</scope>
```
Comment (elahrvivaz) [inline geomesa-process/pom.xml]: was this just redundant? we still want it to be provided scope, right?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala (hunk 1 above)
Comment (jdk2pq) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala]: That works, but then I have to use the setAccessible(true) method on a protected class in order to get the "exists" and "create" methods. I'm not sure which is the better approach, but I'll push up the changes to compare.

File: geomesa-process/pom.xml (hunk 2 above)
Comment (jdk2pq) [inline geomesa-process/pom.xml]: I thought it was redundant since accumulo-core is already marked as provided in GeoMesa.

Comment (elahrvivaz) [inline geomesa-process/pom.xml]: that's fine, just make sure the dependency tree doesn't change.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala (hunk 1 above)
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/AccumuloVersion.scala]: Eh, well it's all kind of a hack, so either way is fine - i'll leave it up to you

File: geomesa-process/pom.xml (hunk 2 above)
Comment (jdk2pq) [inline geomesa-process/pom.xml]: ╭─jkenneally@jkenneally  geomesa-process  ‹fcr_make_acc_15_work› 
╰─$ mvn dependency:tree -Dverbose | grep accumulo-core
[INFO] +- org.apache.accumulo:accumulo-core:jar:1.6.2:provided

looks good

**Comment by anthonyccri** [comment]
b655ede..10be6fe

**Pull request closed without merging**